### PR TITLE
slashing: cleanup

### PR DESCRIPTION
**Pull request by 8sunyuan**
PR Fixes:
- https://github.com/Layr-Labs/eigenlayer-contracts/pull/679#discussion_r1769243035
- https://github.com/Layr-Labs/eigenlayer-contracts/pull/679#discussion_r1769256480 added legacy minWithdrawalDelayBlocks interfaces
- https://github.com/Layr-Labs/eigenlayer-contracts/pull/679#discussion_r1769370145 fixed interface to "numToClear" and added to natspec description
- https://github.com/Layr-Labs/eigenlayer-contracts/pull/679#discussion_r1769374303 addressed, see `modifyAllocations`
- https://github.com/Layr-Labs/eigenlayer-contracts/pull/679#discussion_r1769375520 added helpers using `Rounding.Up` in the OZ Math lib
- https://github.com/Layr-Labs/eigenlayer-contracts/pull/679#discussion_r1772293939 Allow for allocationDelays of 0 now https://github.com/Layr-Labs/eigenlayer-contracts/pull/842/commits/86845acb8da47dc1a69fb8a77a90a93a7dc93d51
- https://github.com/Layr-Labs/eigenlayer-contracts/pull/679#discussion_r1772294865
- https://github.com/Layr-Labs/eigenlayer-contracts/pull/679#discussion_r1772320029 https://github.com/Layr-Labs/eigenlayer-contracts/pull/842/commits/664f9cd076abde998e9d89e31fb5336a85ceb762
- https://github.com/Layr-Labs/eigenlayer-contracts/pull/679#discussion_r1772321327 https://github.com/Layr-Labs/eigenlayer-contracts/pull/842/commits/4d9580f1a0cc95cd9453b3fd6eab9e18720e7768
- https://github.com/Layr-Labs/eigenlayer-contracts/pull/679#discussion_r1772326774
- https://github.com/Layr-Labs/eigenlayer-contracts/pull/679#discussion_r1772326774 fixed natspec
- https://github.com/Layr-Labs/eigenlayer-contracts/pull/679#discussion_r1772327394 nit comment
- https://github.com/Layr-Labs/eigenlayer-contracts/pull/679#discussion_r1772327394
- https://github.com/Layr-Labs/eigenlayer-contracts/pull/679#discussion_r1773850362

Additional Changes:

- totalMagnitude -> maxMagnitude
- renamed `scaledSharesToWithdraw` to `scaledShares`. Previous name indicates that its the actual shares to withdraw which is a bit misleading
- added some clarification to comments and natspec
- IDelegationManager still had description of undelegation limbo in its natspec. This was removed
- fixed `run-coverage` CI due to RPC_MAINNET env not being set

### CODE REVIEW

File: docs/release/slashing/AllocationManager.md [hunk 1]
```diff
@@ -103,7 +103,7 @@ function clearDeallocationQueue(
 ) external;
 ```
 
-This function is used to complete pending deallocations for a list of strategies for an operator. The function takes a list of strategies and the number of pending deallocations to complete for each strategy. For each strategy, the function completes a modification if its effect timestamp has passed. 
+This function is used to complete pending deallocations for a list of strategies for an operator. The function takes a list of strategies and the number of pending deallocations to complete for each strategy. For each strategy, the function completes pending modifications if their effect timestamps have passed.
```
Comment (ypatil12) [inline docs/release/slashing/AllocationManager.md:106]: Nit: `pending deallocations`

File: src/contracts/interfaces/IDelegationManager.sol [hunk 2]
```diff
@@ -134,11 +134,12 @@ interface IDelegationManagerTypes {
         uint32 startTimestamp;
         // Array of strategies that the Withdrawal contains
         IStrategy[] strategies;
-        // TODO: Find a better name for this
-        // Array containing the amount of staker's scaledSharesToWithdraw for withdrawal in each Strategy in the `strategies` array
-        // Note that these shares need to be multiplied by the operator's totalMagnitude at completion to include
-        // slashing occurring during the queue withdrawal delay
-        uint256[] scaledSharesToWithdraw;
+        // Array containing the amount of staker's scaledShares for withdrawal in each Strategy in the `strategies` array
+        // Note that these scaledShares need to be multiplied by the operator's maxMagnitude at completion to include
```
Comment (ypatil12) [inline src/contracts/interfaces/IDelegationManager.sol:138]: Shouldn't we also add beacon chain scaling factor here?

File: src/contracts/core/AllocationManager.sol [hunk 3]
```diff
@@ -256,12 +262,12 @@ contract AllocationManager is
      * @param delay The allocation delay in seconds.
      */
     function _setAllocationDelay(address operator, uint32 delay) internal {
-        require(delay != 0, InvalidAllocationDelay());
```
Comment (ypatil12) [inline src/contracts/core/AllocationManager.sol:259]: Nit: delete this error

File: src/contracts/core/DelegationManager.sol [hunk 4]
```diff
@@ -360,6 +362,35 @@ contract DelegationManager is
         });
     }
 
+    /**
+     *
+     *                         BACKWARDS COMPATIBLE LEGACY FUNCTIONS
+     *                         TO BE DEPRECATED IN FUTURE
+     *
```
Comment (ypatil12) [inline src/contracts/core/DelegationManager.sol:369]: I feel like we may want a non-reverting `getCompletableTimestamp` function? Wdyt

File: src/contracts/core/DelegationManager.sol [hunk 5]
```diff
@@ -544,14 +575,29 @@ contract DelegationManager is
...
+     * @param strategies The strategies to queue a withdrawal for
+     * @param sharesToWithdraw The amount of shares the staker wishes to withdraw, must be less than staker's withdrawable shares
+     * @param maxMagnitudes The corresponding maxMagnitudes of the operator for the respective strategies
+     *
+     * @dev The amount withdrawable by the staker may not actually be the same as the depositShares that are in storage in the StrategyManager/EigenPodManager.
+     * This is a result of any slashing that has occurred during the time the staker has been delegated to an operator. So the proportional amount that is withdrawn
+     * out of the amount withdrawable for the staker has to also be decremented from the staker's deposit shares.
+     * Ex. Suppose as a staker, I have 100 depositShares for a strategy thats sitting in the StrategyManager in the `stakerDepositShares` mapping but I actually have been slashed 50%
+     * and my real withdrawable amount is 50 shares.
+     * Now when I go to withdraw 20 shares, I'm propotionally withdrawing 40% of my withdrawable shares. We calculate below via the `toDepositShares()` function to
```
Comment (ypatil12) [inline src/contracts/core/DelegationManager.sol:594]: Nit: proportionally spelling

File: src/contracts/libraries/SlashingLib.sol [hunk 6]
```diff
@@ -47,6 +47,15 @@ library SlashingLib {
         return x.mulDiv(WAD, y);
     }
 
+    /**
+     * @notice Used explicitly for calculating slashed magnitude, we want to ensure even in the
+     * situation where an operator is slashed several times and precision has been lost over time,
+     * an incoming slashing request isn't rounded down to 0 and an operator is able to avoided slashing penalties.
```
Comment (ypatil12) [inline src/contracts/libraries/SlashingLib.sol:53]: nit: able to avoid

**Review by ypatil12 — COMMENTED** [review]
Generally looks good. Can we add explicit tests on rounding up?

File: src/contracts/libraries/SlashingLib.sol (hunk 6 above)
Comment (8sunyuan) [reply src/contracts/libraries/SlashingLib.sol]: https://github.com/Layr-Labs/eigenlayer-contracts/pull/842/commits/ba893331d545b5feb09573287e5f37b1fc63a27c

File: src/contracts/core/DelegationManager.sol (hunk 5 above)
Comment (8sunyuan) [reply src/contracts/core/DelegationManager.sol]: https://github.com/Layr-Labs/eigenlayer-contracts/pull/842/commits/ba893331d545b5feb09573287e5f37b1fc63a27c

File: src/contracts/core/AllocationManager.sol (hunk 3 above)
Comment (8sunyuan) [reply src/contracts/core/AllocationManager.sol:259]: https://github.com/Layr-Labs/eigenlayer-contracts/pull/842/commits/ba893331d545b5feb09573287e5f37b1fc63a27c

File: src/contracts/interfaces/IDelegationManager.sol (hunk 2 above)
Comment (8sunyuan) [reply src/contracts/interfaces/IDelegationManager.sol]: https://github.com/Layr-Labs/eigenlayer-contracts/pull/842/commits/ba893331d545b5feb09573287e5f37b1fc63a27c#diff-d3c92783ec657b412dca73e18454a61feff67efa9b92f6e72a48eaa863a2626a
Added note on that

File: docs/release/slashing/AllocationManager.md (hunk 1 above)
Comment (8sunyuan) [reply docs/release/slashing/AllocationManager.md]: https://github.com/Layr-Labs/eigenlayer-contracts/pull/842/commits/ba893331d545b5feb09573287e5f37b1fc63a27c#diff-f1b6b904827b60cc038619a4d17fef5e1e31338648dd3fe52168825e8fd6a67c

**Review by ypatil12 — APPROVED** [review]

**Review by ypatil12 — APPROVED** [review]

**Pull request merged**